src/pages/workspace/workflows/WorkspaceWorkflowsApproverPage.tsx

```diff
@@ -0,0 +1,201 @@
+import {withOnyx} from 'react-native-onyx';
+import type {OnyxEntry} from 'react-native-onyx';
+import Badge from '@components/Badge';
+import FullPageNotFoundView from '@components/BlockingViews/FullPageNotFoundView';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import ScreenWrapper from '@components/ScreenWrapper';
+import SelectionList from '@components/SelectionList';
+import type {ListItem, Section} from '@components/SelectionList/types';
+import UserListItem from '@components/SelectionList/UserListItem';
+import useLocalize from '@hooks/useLocalize';
+import useNetwork from '@hooks/useNetwork';
+import useStyleUtils from '@hooks/useStyleUtils';
+import useThemeStyles from '@hooks/useThemeStyles';
+import compose from '@libs/compose';
+import {formatPhoneNumber} from '@libs/LocalePhoneNumber';
+import Log from '@libs/Log';
+import * as LoginUtils from '@libs/LoginUtils';
+import Navigation from '@libs/Navigation/Navigation';
+import * as PersonalDetailsUtils from '@libs/PersonalDetailsUtils';
+import {parsePhoneNumber} from '@libs/PhoneNumber';
+import * as PolicyUtils from '@libs/PolicyUtils';
+import * as UserUtils from '@libs/UserUtils';
+import withPolicyAndFullscreenLoading from '@pages/workspace/withPolicyAndFullscreenLoading';
+import type {WithPolicyAndFullscreenLoadingProps} from '@pages/workspace/withPolicyAndFullscreenLoading';
+import * as Policy from '@userActions/Policy';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {PersonalDetailsList, PolicyMember} from '@src/types/onyx';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+
+type WorkspaceWorkflowsApproverPageOnyxProps = {
+    /** All of the personal details for everyone */
+    personalDetails: OnyxEntry<PersonalDetailsList>;
+};
+
+type WorkspaceWorkflowsApproverPageProps = WorkspaceWorkflowsApproverPageOnyxProps & WithPolicyAndFullscreenLoadingProps;
+type MemberOption = Omit<ListItem, 'accountID'> & {accountID: number};
+type MembersSection = SectionListData<MemberOption, Section<MemberOption>>;
+
+function WorkspaceWorkflowsApproverPage({policy, policyMembers, personalDetails, isLoadingReportData = true}: WorkspaceWorkflowsApproverPageProps) {
+    const {translate} = useLocalize();
+    const policyName = policy?.name ?? '';
+    const [searchTerm, setSearchTerm] = useState('');
+    const {isOffline} = useNetwork();
+    const styles = useThemeStyles();
+    const StyleUtils = useStyleUtils();
+
+    const isDeletedPolicyMember = (policyMember: PolicyMember): boolean =>
+        !isOffline && policyMember.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE && isEmptyObject(policyMember.errors);
+
+    const [formattedPolicyMembers, formattedApprover] = useMemo(() => {
+        const policyMemberDetails: MemberOption[] = [];
+        const approverDetails: MemberOption[] = [];
+
+        Object.entries(policyMembers ?? {}).forEach(([accountIDKey, policyMember]) => {
```

Comment:
This seems to be a copy/pasta from https://github.com/Expensify/App/blob/59f8f3d4889afe26c3f73c2e8e3df1ced50e11dd/src/pages/workspace/WorkspaceMembersPage.tsx#L282. Can we just extract that into a method and reuse it in both places?

Reply:
@luacmartins i agree that this method shares most of the code. But I don't think it's a good idea to create a method that would reuse common logic.

1. Both methods in the page are unrelated and DRYing would couple them together. So it'll be harder to change them later
2. The two methods are different in the following ways: 
     1. approverpage: pushes the formatted option to 2 arrays. members page pushes to 1 array. 
     2. The disabled logic for both pages is different. 


This is the only common code - I'm not sure what to name this or how would this be reused in other places.
What do you think?

```js

const accountID = Number(accountIDKey);
if (isDeletedPolicyMember(policyMember)) {
    return;
}

const details = personalDetails?.[accountID];

if (!details) {
    Log.hmmm(`[WorkspaceMembersPage] no personal details found for policy member with accountID: ${accountID}`);
    return;
}

if (PolicyUtils.isExpensifyTeam(details?.login ?? details?.displayName)) {
    if (policyOwner && currentUserLogin && !PolicyUtils.isExpensifyTeam(policyOwner) && !PolicyUtils.isExpensifyTeam(currentUserLogin)) {
        return;
    }
}

const isOwner = policy?.owner === details.login;
const isAdmin = session?.email === details.login || policyMember.role === CONST.POLICY.ROLE.ADMIN;

let roleBadge = null;
if (isOwner || isAdmin) {
    roleBadge = (
        <Badge
            text={isOwner ? translate('common.owner') : translate('common.admin')}
            textStyles={styles.textStrong}
            badgeStyles={[styles.justifyContentCenter, StyleUtils.getMinimumWidth(60), styles.badgeBordered]}
        />
    );
}
```